pkg/kubelet/containerdshim/containerd_container.go

```diff
@@ -151,26 +173,39 @@ func (cs *containerdService) StartContainer(containerID string) error {
 	if containerID == "" {
 		return fmt.Errorf("containerID should not be empty")
 	}
-	if _, err := cs.cdClient.Start(gocontext.Background(), &execution.StartRequest{
-		ID: containerID,
-	}); err != nil {
-		glog.V(2).Infof("StartContainer for %s failed with %v", containerID, err)
-		return err
-	}
-	return nil
+	_, err := cs.cdClient.Start(gocontext.Background(), &execution.StartRequest{ID: containerID})
+	return err
 }
 
 // StopContainer stops a running container with a grace period (i.e., timeout).
 // P0
 func (cs *containerdService) StopContainer(containerID string, timeout int64) error {
-	return fmt.Errorf("not implemented")
+	glog.V(2).Infof("StopContainer called with %s", containerdID)
+	if containerID == "" {
+		return fmt.Errorf("containerID should not be empty")
+	}
+	c, err := getShimClient(containerID)
+	if err != nil {
+		return fmt.Errorf("failed to get shim for container %q: %v", containerID, err)
+	}
+	_, err := c.Exit(gocontext.Background(), &shim.ExitRequest{})
```

Comment:
I believe exiting a new shim won't help unless that's the shim we used to start the process? Would have to test it I guess.